examples/AvroGeneric/Program.cs

```diff
@@ -0,0 +1,121 @@
+                // the Java implementation.
+                { "schema.registry.url", schemaRegistryUrl },
+                // optional schema registry client properties:
+                { "schema.registry.connection.timeout.ms", 5000 },
+                { "schema.registry.max.cached.schemas", 10 },
+                // optional avro serializer properties:
+                { "avro.serializer.buffer.bytes", 50 },
+                { "avro.serializer.auto.register.schemas", true }
+            };
+
+            var consumerConfig = new Dictionary<string, object>
+            {
+                { "bootstrap.servers", bootstrapServers },
+                { "group.id", Guid.NewGuid() },
+                { "schema.registry.url", schemaRegistryUrl }
+            };
+
+            var s = (RecordSchema)Schema.Parse(
+                @"{
+                    ""namespace"": ""Confluent.Kafka.Examples.AvroSpecific"",
+                    ""type"": ""record"",
+                    ""name"": ""User"",
+                    ""fields"": [
+                        {""name"": ""name"", ""type"": ""string""},
+                        {""name"": ""favorite_number"",  ""type"": [""int"", ""null""]},
+                        {""name"": ""favorite_color"", ""type"": [""string"", ""null""]}
+                    ]
+                  }"
+            );
+            
+            using (var consumer = new Consumer<string, GenericRecord>(consumerConfig, new AvroDeserializer<string>(), new AvroDeserializer<GenericRecord>()))
+            using (var producer = new Producer<string, GenericRecord>(producerConfig, new AvroSerializer<string>(), new AvroSerializer<GenericRecord>()))
+            {
+                consumer.OnMessage += (o, e)
+                    => Console.WriteLine($"Key: {e.Key}\nValue: {e.Value}");
+
+                consumer.OnError += (_, e)
+                    => Console.WriteLine("Error: " + e.Reason);
+
+                consumer.OnConsumeError += (_, e)
+                    => Console.WriteLine("Consume error: " + e.Error.Reason);
+
+                consumer.Subscribe(topicName);
+
+                CancellationTokenSource cts = new CancellationTokenSource();
+                var consumeTask = Task.Factory.StartNew(() =>
+                {
+                    while (!cts.Token.IsCancellationRequested)
+                    {
+                        consumer.Poll(100);
+                    }
+                });
+                
+                Console.WriteLine($"{producer.Name} producing on {topicName}. Enter user names, q to exit.");
+
+                int i = 0;
+                string text;
+                while ((text = Console.ReadLine()) != "q")
+                {
+                    var record = new GenericRecord(s);
+                    record.Add("name", text);
+                    record.Add("favorite_number", i++);
+                    record.Add("favorite_color", "blue");
```

Comment:
so many u's missing hereI don't even know where to start :)

Reply:
i'm numb to this by now